lib/datetime/datetime.ex

```diff
@@ -792,6 +792,12 @@ defmodule Timex.DateTime do
       m < 0  -> %{datetime | :year => year + div(m, 12), :month => 12 + rem(m, 12)}
       :else  -> %{datetime | :month => m}
     end
+
+    # setting months to remainders may result in month = 0
+    if shifted.month == 0 do
+      shifted = %{ shifted | :year => shifted.year - 1, :month => 12 }
```

Comment:
This kind of imperative-style binding code doesn't work in newer versions of Elixir. Could you change it to:

```elixir
shifted = case shifted.month do
  0 -> %{ shifted | :year => shifted.year - 1, :month => 12 }
  _ -> shifted
end
```